inspirehep/records/api/base.py

```diff
@@ -1030,7 +1030,12 @@ def _update_refs_in_citation_table(self, save_every=100):
         """
         RecordCitations.query.filter_by(citer_id=self.id).delete()
 
-        if self.is_superseded() or self.get("deleted") or self.pid_type not in ["lit"]:
+        if (
+            self.is_superseded()
+            or self.get("deleted")
+            or self.pid_type not in ["lit"]
+            or "Literature" not in self["_collections"]
+        ):
```

Comment:
is this the right place? what happens if a record goes from being not deleted to deleted? does its citations get removed correctly? (if so the docstring is wrong).

Reply:
what about making a `can_cite` property instead of inlining this?